rbd/callback_shims_mimic.go

```diff
@@ -0,0 +1,15 @@
+// +build !luminous
+
+package rbd
+
+/*
+
+#include <rbd/librbd.h>
+
+extern void imageWatchCallback(int index);
+
+void callWatchCallback(int index) {
```

Comment:
The real implementation is defined on the Go side, and so when I implemented our first callback support (in diff_iterate) I was following some guides pretty closely. A little while back I realized there's a simplification that we can probably do because we are creating these "wrap_<ceph_name>" function calls. However, I didn't want to do it in this PR I wanted to update both diff_iterate and this at once later.
Let me know if you want me to elaborate further on this here.

Reply:
Yeah, I would love to understand that better, but maybe not here.